datasets/peer_read/peer_read.py

```diff
@@ -0,0 +1,306 @@
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        url = _URLs["dataset_repo"]
+        data_dir = dl_manager.download_and_extract(url)
+        paths = self._get_paths(
+            data_dir=data_dir,
+            domain=self.config.name,
+        )
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepaths": paths["train"],
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepaths": paths["test"], "split": "test"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepaths": paths["dev"],
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepaths, split):
+        """ Yields examples. """
+        for id_, filepath in enumerate(filepaths):
+            with open(filepath, encoding="utf-8", errors="replace") as f:
+                data = json.load(f)
+                if self.config.name == "parsed_pdfs":
+                    metadata = data.get(
+                        "metadata",
+                        {
+                            "source": "",
+                            "authors": [],
+                            "title": [],
+                            "sections": [],
+                            "references": [],
+                            "referenceMentions": [],
+                            "year": "",
+                            "abstractText": "",
+                            "creator": "",
+                        },
+                    )
+                    metadata["sections"] = [] if metadata["sections"] is None else metadata["sections"]
+                    metadata["sections"] = [
+                        {
+                            "heading": self._decode(section.get("heading", "")),
+                            "text": self._decode(section.get("text", "")),
+                        }
+                        for section in metadata["sections"]
+                    ]
+                    metadata["references"] = [] if metadata["references"] is None else metadata["references"]
+                    metadata["references"] = [
+                        {
+                            "title": reference.get("title", ""),
+                            "author": reference.get("author", []),
+                            "venue": reference.get("venue", ""),
+                            "citeRegEx": reference.get("citeRegEx", ""),
+                            "shortCiteRegEx": reference.get("shortCiteRegEx", ""),
+                            "year": reference.get("year", ""),
```

Comment:
Why use `get` here ?
All these fields have missing values ?

Reply:
Yes some of the datasets don't have some of the fields